### PR TITLE
Functionality to mark folder nodes as "root"

### LINKED ISSUE #454: TreeControl: Globe icon being shown on folders in the local branch list
Only the remote branch list should show a globe for the top level folder nodes (which represent the remotes), but apparently the local branch is also showing it instead of using the normal folder icon. Probably bad copy paste

image

### PR DESCRIPTION
**Pull request by StanleyGoldman**
**Note:** This branch targets `enhancements/branches-view-rollup` #464 

Fixes #454

- Renaming icons to be something more useful
- Adding functionality to render the globe icon at the right time

Depends on:
- [x] #151 
- [x] #465
- [x] #479

### CODE REVIEW

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs [hunk 1]
```diff
@@ -134,19 +134,23 @@ private void Render()
...
         private void BuildTree()
         {
+            if (treeLocals == null)
+            {
+                treeLocals = new Tree();
+                treeLocals.ActiveNodeIcon = Styles.ActiveBranchIcon;
+                treeLocals.NodeIcon = Styles.BranchIcon;
+                treeLocals.FolderIcon = Styles.FolderIcon;
+
+                treeRemotes = new Tree();
```
Comment (sguthals) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs]: treeLocals and treeRemotes are always either defined or null together right? It isn't worth having a separate if for treeRemotes?

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs [hunk 2]
```diff
@@ -383,7 +385,7 @@ private void ResetNodeIcons(TreeNode node)
             }
             else if (node.IsFolder)
             {
-                if (node.Level == 1)
+                if (node.Level == RootFolderIconLevel)
```
Comment (sguthals) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs]: Why was this switched from 1 to -1?

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs (hunk 1 above)
Comment (StanleyGoldman) [reply src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs]: They always get instantiated together, so I figured I could save the `boolean` check.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs (hunk 2 above)
Comment (StanleyGoldman) [reply src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs]: You know what. It's a rudimentary fix for something that i'm already having to change.
I'm going to send a PR before this one that provides a correct solution instead.

Comment (sguthals) [reply src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs]: Sounds good

**Pull request merged**